### PR TITLE
improve magic command completions

### PR DESCRIPTION
**Pull request by jonsequitur**
This fixes a number of cases with magic command completions.

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive/Directives/KernelActionDirective.cs [hunk 1]
```diff
@@ -81,15 +82,28 @@ private set
...
         var subcommandCompletions = Subcommands.Select(s => new CompletionItem(s.Name, WellKnownTags.Method)
         {
             AssociatedSymbol = s,
             Documentation = s.Description
         });
 
-        return subcommandCompletions.Concat(baseCompletions).ToArray();
+        completions.AddRange(subcommandCompletions);
+
+        if (Parent is KernelDirective parent)
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive/Directives/KernelActionDirective.cs:97]: Does order in which the completions are added to the list matter? If yes, should the directive-based ones be added first?

File: src/Microsoft.DotNet.Interactive/Parsing/PolyglotSyntaxParser.cs [hunk 2]
```diff
@@ -188,12 +188,16 @@ void AppendNode(SyntaxNode node)
...
                     case DirectiveParameterValueNode valueNode when subcommandNode is null:
                         directiveNode.Add(valueNode);
                         break;
+                    case DirectiveParameterValueNode valueNode when subcommandNode is not null:
+                        subcommandNode.Add(valueNode);
+                        break;
 
-                    case DirectiveNameNode nameNode when subcommandNode is not null:
-                        subcommandNode.Add(nameNode);
+                    case DirectiveParameterNode parameterNode when subcommandNode is null:
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive/Parsing/PolyglotSyntaxParser.cs:199]: Is this a duplicate of the `case` on line 188 above?

**Review by shyamnamboodiripad — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive/Directives/KernelActionDirective.cs (hunk 1 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive/Directives/KernelActionDirective.cs:97]: It doesn't matter. At this level, the final list hasn't been assembled, so ordering doesn't matter yet, nor does e.g. filtering for partially-typed completions.

File: src/Microsoft.DotNet.Interactive/Parsing/PolyglotSyntaxParser.cs (hunk 2 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive/Parsing/PolyglotSyntaxParser.cs:199]: It is! Nice catch.

**Pull request merged**